core/src/test/java/org/infinispan/test/MultipleCacheManagersTest.java

```diff
@@ -420,8 +425,26 @@ protected DataContainer dataContainer(int index) {
       return advancedCache(index).getDataContainer();
    }
 
-   @Factory
+   /**
+    * This is the method you should override when providing factory method.
+    */
    public Object[] factory() {
+      throw new IllegalStateException("Only overridden methods should be called!");
+   }
+
+   @Factory
+   public Object[] defaultFactory() {
+      // It is possible to override the factory method, but if we extend a class that defines such overridden
+      // method, the factory method will be inherited, too - that results in running the superclass tests
```

Comment:
I personally don't like it. I think our injection framework should somehow be fixed to do what we need (instead of doing tricks like this).